packages/ui-settings/src/types.ts

```diff
@@ -28,3 +29,19 @@ export interface NetworkSpecsStruct {
   title: string;
   unit: string;
 }
+
+export type Endpoint = JsonRpcEndpoint | SubstrateConnectEndpoint;
+export interface JsonRpcEndpoint {
+  type: EndpointType.jrpc;
+  url: string;
+}
+
+export interface SubstrateConnectEndpoint {
+  type: EndpointType.substrateconnect;
+  chain: string;
+}
+
+export enum EndpointType {
```

Comment:
Remove the enum, it is hugely problematic -

- it creates a bunch of code to make this work
- it turns type-only imports into type & code imports

This is sufficient: `export type EndpointType = 'json-rpc' | 'subscrate-connect';`